File: crypto/src/main/java/io/oversec/one/crypto/encoding/pad/PadderContent.java
package io.oversec.one.crypto.encoding.pad;

import java.util.Comparator;

public class PadderContent {
    private long mKey;
    private String mName;
    private String mSort;
    private String mContent;

    public static Comparator<? super PadderContent> sortComparator = new Comparator<PadderContent>() {
        @Override
        public int compare(PadderContent lhs, PadderContent rhs) {
            return lhs.mSort.compareTo(rhs.mSort);
        }
    };

    public PadderContent() {
    }

    public PadderContent(String name, String content) {
        mKey = System.currentTimeMillis();
        mSort = name;
        mName = name;
        mContent = content;


    }


    public PadderContent(String sort, String name, String content) {
        mKey = System.currentTimeMillis();
        mSort = sort;
        mName = name;
        mContent = content;


    }

    public String getName() {
        return mName;
    }

    public void setName(String mName) {
        this.mName = mName;
    }

    public String getContent() {
        return mContent;
    }

    public void setContent(String mContent) {
        this.mContent = mContent;
    }

    public long getKey() {
        return mKey;
    }

    public String getContentBegin() {
        final int length = 30;
        return mContent.length() < 30 ? mContent : mContent.substring(0, length);
    }

    public void setSort(String v) {
        mSort = v;
    }
}

File: crypto/src/main/java/io/oversec/one/crypto/encoding/pad/PadderContent.kt
package io.oversec.one.crypto.encoding.pad

import java.util.Comparator

class PadderContent {
    var key: Long = 0
        private set
    lateinit var name: String
    private lateinit  var mSort: String
    lateinit var content: String

    val contentBegin: String
        get() {
            val length = 30
            return if (content.length < 30) content else content.substring(0, length)
        }

    /*required empty constructor for waspDb*/
    constructor()

    constructor(name: String, content: String) {
        key = System.currentTimeMillis()
        mSort = name
        this.name = name
        this.content = content
    }

    constructor(sort: String, name: String, content: String) {
        key = System.currentTimeMillis()
        mSort = sort
        this.name = name
        this.content = content
    }

    fun setSort(v: String) {
        mSort = v
    }

    companion object {
        var sortComparator: Comparator<in PadderContent> =
            Comparator { lhs, rhs -> lhs.mSort.compareTo(rhs.mSort) }
    }
}
